### ISSUE TITLE
Support proper chapter names (content_metadata)

### ISSUE THREAD

**Issue by katyavera** [KrumpetPirate/AAXtoMP3#160]
It would be really nice to support the descriptive chapter names that are shown in Audible.

At least on Windows, with `Audible for Windows`, chapter names are available under `%LOCALAPPDATA%\Packages\AudibleInc.AudibleforWindowsPhone_xns73kv1ymhp2\LocalState\filescache` .

I've created a script to convert the JSON to the format supported by `mp4chaps`.
Note: In the aax file (and the current m4b file) there aren't so many chapters. This script creates a chapter for every declared chapter in the json file (which is what I would expect).

I haven't integrated it with the bash script yet.

``` python
#!/usr/bin/env python3

import json
import math
import os
import sys


def print_chapters(data):
    try:
        chapters = data["content_metadata"]["chapter_info"]["chapters"]
    except KeyError:
        print("content_metadata.chapter_info.chapters missing from JSON",
              file=sys.stderr)
        sys.exit(1)

    # Used to creating for example CHAPTER01= or CHAPTER001=, depending how many chapters there are
    chapters_placeholder = "0" + str(1 + int(math.log(len(chapters), 10)))

    for i, chapter in enumerate(chapters):
        n = i + 1  # output from CHAPTER01, not CHAPTER00
        h = int(chapter["start_offset_ms"] / 1000 / 3600)
        m = int(chapter["start_offset_ms"] / 1000 / 60) % 60
        sec = int(chapter["start_offset_ms"] / 1000) % 60
        ms = chapter["start_offset_ms"] % 1000
        prefix = ("CHAPTER{:" + chapters_placeholder + "d}").format(n)
        print("{}={:02d}:{:02d}:{:02d}.{:03d}".format(prefix, h, m, sec, ms))
        print("{}NAME={}".format(prefix, chapter["title"]))


def main():
    if len(sys.argv):
        print("Usage: {} content_metadata_XXXXXXXXX.json".format(
            sys.argv[0]), file=sys.stderr)
        print("", file=sys.stderr)
        print("Find the content_medata json files containing chapter information under", file=sys.stderr)
        print(os.path.expandvars(
            r"  %LOCALAPPDATA%\Packages\AudibleInc.AudibleforWindowsPhone_xns73kv1ymhp2\LocalState\filescache"), file=sys.stderr)
        sys.exit(1)

    filename = sys.argv[1]

    try:
        # At least "Audible for Windows" creates json files that start with a byte order mark (BOM)
        with open(filename, 'r', encoding='utf-8-sig') as f:
            data = json.load(f)
    except UnicodeDecodeError:
        # Open a "normal" json
        with open(filename, 'r') as f:
            data = json.load(f)

    print_chapters(data)


if __name__ == '__main__':
    main()
```

Expected output:
```
CHAPTER001=00:00:00.000
CHAPTER001NAME=Preface
CHAPTER002=00:13:54.107
CHAPTER002NAME=PART ONE - Before the State
...
CHAPTER158=22:27:12.804
CHAPTER158NAME=What Comes Next
```

**Comment by fabh2o** [comment]
I was looking for the same functionality, then i found out about https://github.com/mkb79/audible-cli (it basically have the ability to download more infos from audible such as chapter titles). I need to do some final things before opening a pull request, but this might help you: https://github.com/fabh2o/AAXtoMP3/tree/audible-cli-integration#audible-cli-integration.

A rough step-by-step:
```bash
# Install audible-cli
git clone https://github.com/mkb79/audible-cli.git
cd audible-cli
pip install .

# Interactive wizard to set up the config
audible-quickstart

# Clone the repo and use the proper branch
git clone https://github.com/fabh2o/AAXtoMP3.git
cd AAXtoMP3
git checkout audible-cli-integration

# Download files, dont change the names
mkdir -v Source
cd Source
audible download --all --aax --cover --cover-size 1215 --chapter

# Convert and use the new infos (chapter title + HD cover)
cd ..
./AAXtoMP3 --use-audible-cli-data -e:m4b -c Source/*.aax
```
Note that you need to have `jq` installed to parse the json files

**Comment by BigCheeZ** [comment]
@fabh2o thanks for this. It was very helpful. FYI, I added support for using this same JSON info when using a single output file instead of splitting by chapter.

https://github.com/fabh2o/AAXtoMP3/pull/1

**Comment by fabh2o** [comment]
@KrumpetPirate I think that now with the new version of the script this issue can be closed  #171
